libbpf-rs/src/map.rs

```diff
@@ -1203,40 +1213,112 @@ bitflags! {
+    /// A map in map storage.
+    /// One level of nesting is supported, where an outer map contains instances of a single type
+    /// of inner map. Refer the kernel [documentation](https://docs.kernel.org/bpf/map_of_maps.html) for more details.
     HashOfMaps = libbpf_sys::BPF_MAP_TYPE_HASH_OF_MAPS,
+    /// An array map that uses the key as the index to lookup a reference to a net device.
+    /// Primarily used for XDP BPF Helper.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/map_devmap.html) for more details.
     Devmap = libbpf_sys::BPF_MAP_TYPE_DEVMAP,
+    /// An array map holds references to a socket descriptor.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/map_sockmap.html) for more details.
     Sockmap = libbpf_sys::BPF_MAP_TYPE_SOCKMAP,
+    /// A map that redirects raw XDP frames to another CPU.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/map_cpumap.html) for more details.
     Cpumap = libbpf_sys::BPF_MAP_TYPE_CPUMAP,
+    /// A map that redirects raw XDP frames to AF_XDP sockets (XSKs), a new type of address family
+    /// in the kernel that allows redirection of frames from a driver to user space without
+    /// having to traverse the full network stack. Refer the kernel [documentation](https://docs.kernel.org/bpf/map_xskmap.html) for more details.
     Xskmap = libbpf_sys::BPF_MAP_TYPE_XSKMAP,
+    /// A Hash map that holds references to sockets via their socket descriptor.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/map_sockmap.html) for more details.
     Sockhash = libbpf_sys::BPF_MAP_TYPE_SOCKHASH,
+    /// Deprecated. Use `CGrpStorage` instead.
+    ///
+    /// A Local storage for cgroups.
+    /// Only available with `CONFIG_CGROUP_BPF` and to programs that attach to cgroups.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/cgroup_storage.html) for more details.
     CgroupStorage = libbpf_sys::BPF_MAP_TYPE_CGROUP_STORAGE,
+    /// A Local storage for cgroups.  
+    /// Only available with `CONFIG_CGROUPS`.
+    /// Refer the kernel [documentation](https://docs.kernel.org/bpf/cgroup_storage.html) for more details.
+    ///
+    /// See also [Difference between cgrp_storage and cgroup_storage](https://docs.kernel.org/bpf/map_cgrp_storage.html#difference-between-bpf-map-type-cgrp-storage-and-bpf-map-type-cgroup-storage)
+    CGrpStorage = libbpf_sys::BPF_MAP_TYPE_CGRP_STORAGE,
+    /// A map that holds references to sockets with `SO_REUSEPORT` option set.
+    /// Refer [documentation](https://docs.ebpf.io/linux/map-type/BPF_MAP_TYPE_REUSEPORT_SOCKARRAY/) for more details.
     ReuseportSockarray = libbpf_sys::BPF_MAP_TYPE_REUSEPORT_SOCKARRAY,
+    /// A per-CPU variant of [`BPF_MAP_TYPE_CGROUP_STORAGE`][`MapType::CgroupStorage`].
+    /// Refer [documentation](https://docs.ebpf.io/linux/map-type/BPF_MAP_TYPE_PERCPU_CGROUP_STORAGE) for more details.
```

Comment:
Do we have to fall back to `docs.ebpf.io` because the kernel documentation doesn't contain documentation for this item? What a mess :-(